conda/core/portability.py

```diff
@@ -28,7 +28,7 @@
                  br'(.*)'  # the rest of the line can contain option flags
                  br')$')  # end whole_shebang group
 
-MAX_SHEBANG_LENGTH = 127 if on_linux else 512  # Not used on Windows
+MAX_SHEBANG_LENGTH = 512 if on_win else 127
```

Comment:
Windows has no concept of shebang, so if the intent here is to have 127 for both Linux and Mac, I'd say we keep it simple:

```suggestion
MAX_SHEBANG_LENGTH = 127
```

Reply:
Ah, in which case I think we all misunderstood the purpose of this code then, this change shouldn't be merged.

cc @jezdez @travishathaway

Reply:
@kenodegard, I think this still needs to be merged. My suggestion would be to go with what @jaimergp put forth. It fixes a bug that was found during a build test.